apps/server/src/services/RestoreService.ts

```diff
@@ -0,0 +1,124 @@
+import { LogOrigin, Playback } from 'ontime-types';
+import { logger } from '../classes/Logger.js';
+import { PlaybackService } from './PlaybackService.js';
+import { eventTimer } from './TimerService.js';
+import { EventLoader } from '../classes/event-loader/EventLoader.js';
+
+//File stuff
+import { tmpdir } from 'os'
+import fs from 'fs';
+import path from 'path';
+import { Writer } from 'steno';
+
+
+/**
+ * Service manages saveing of timer state
+ * that can then be resored when reopening
+ */
+class RestoreService {
+    private lastStore: string = '';
+    private readonly file;
+    private playback: Playback;
+    private selectedEventId: string | null;
+    private startedAt: number | null;
+    private addedTime: number | null;
+    private pausedAt: number | null;
+    private readonly hasValidData;
+    private readonly filePath;
+
+    constructor() {
+        this.filePath = path.join(tmpdir(), 'ontime');
+        if (!fs.existsSync(this.filePath)) {
+            try {
+                fs.mkdirSync(this.filePath);
+            } catch (err) {
+                logger.error(LogOrigin.Server, err);
+                this.hasValidData = false;
+                return;
+            }
+        }
+
+        try {
+            const data = fs.readFileSync(path.join(this.filePath, 'restore.csv'), 'utf-8');
+            const elements = data.split(',');
+            if (elements[5] == '\n') {
+                logger.info(LogOrigin.Server, 'intact restore file');
+
+                const maybePlayback = elements[0] as Playback
+                if (!Object.values(Playback).includes(maybePlayback)) { this.hasValidData = false; return; }
+                this.playback = maybePlayback;
+
+                const maybeId = elements[1] === 'null' ? null : elements[1];
+                //TODO: Cannot access 'EventLoader' before initialization
+                // if (maybeId && EventLoader.getEventWithId(maybeId)) { this.hasValidData = false; return; }
```

Comment:
is this something I need to worry about? or is it resolved?

Reply:
Yes I'm working on this check
And this first try did not work do to that error
If you have a way to fix it that would be nice 
Otherwise I will try to find  another way

Reply:
I see, I dont know if this is the issue, but it seems that you are trying to call a static property on the class EventLoader rather than on the instance of eventLoader, 
so it should be 
`eventLoader.getEventWithId `// note the lower case E

Reply:
I did try that
Property 'getEventWithId' does not exist on type 'EventLoader'. Did you mean to access the static member 'EventLoader.getEventWithId' instead?ts(2576)

Reply:
hm, you are correct. ignore me, it is a static method.
In that case, you need to ensure the correct order of initialisation as the error suggests

It sort of looks like that should happen according to your code, looks like this might be a race condition
```ts
  // provide initial payload to event store
  eventLoader.init();
  eventStore.init(getInitialPayload());
  restoreService.restore();
```

Either way, I am unsure if loading the event should be of the scope of the service. as we mentioned before, the restore sequence could return to the app a possibly restorable state and let the app deal with it.

in turn, the app would ask the playback service to load an event, if that doesn't exist it would fail

Reply:
> in turn, the app would ask the playback service to load an event, if that doesn't exist it would fail

Ok I can live with this